src/NvGet.Tools.Shared/Arguments/ConsoleArgsContext.cs

```diff
@@ -142,5 +143,40 @@ async Task<IEnumerable<UpdateResult>> LoadFromStreamAsync()
 				}
 			}
 		}
+
+		internal static IEnumerable<(string PropertyName, string PackageId)> LoadProjectProperties(string inputPathOrUrl)
+		{
+			var results =
+				LoadFromStreamAsync()
+					.GetAwaiter()
+					.GetResult();
+			return results;
+
+			async Task<IEnumerable<(string PropertyName, string PackageId)>> LoadFromStreamAsync()
+			{
+				var jsonSerializer = JsonSerializer.CreateDefault();
+
+				if(inputPathOrUrl.StartsWith("http://") || inputPathOrUrl.StartsWith("https://"))
+				{
+					using(var httpClient = new HttpClient())
```

Comment:
There should be a single static HttpClient used both here and in LoadOverrides. See https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-7.0#instancing:

> HttpClient is intended to be instantiated once and reused throughout the life of an application. In .NET Core and .NET 5+, HttpClient pools connections inside the handler instance and reuses a connection across multiple requests. If you instantiate an HttpClient class for every request, the number of sockets available under heavy loads will be exhausted. This exhaustion will result in SocketException errors.